src/main/java/org/mitre/synthea/helpers/RandomCodeGenerator.java

```diff
@@ -26,166 +36,91 @@
+  public static String expandBaseUrl = Config.get("generate.terminology_service_url") + "/ValueSet/$expand?url=";
 
   private static final Logger logger = LoggerFactory.getLogger(RandomCodeGenerator.class);
-  private static TerminologyClient terminologyClient = null;
-  private static LoadingCache<ExpandInput, ValueSet> responseCache;
-
-  static {
-    String terminologyServiceUrl = Config.get("generate.terminology_service_url");
-    initialize(terminologyServiceUrl);
-  }
-
-  /**
-   * Initialize the RandomCodeGenerator with the supplied terminology service URL.
-   *
-   * @param terminologyServiceUrl the URL of the FHIR terminology service to be used
-   */
-  public static void initialize(String terminologyServiceUrl) {
-    if (terminologyServiceUrl != null && !terminologyServiceUrl.isEmpty()) {
-      terminologyClient = FhirContext.forR4()
-          .newRestfulClient(TerminologyClient.class, terminologyServiceUrl);
-      initializeCache();
-    }
-  }
+  public static Map<String, List<Object>> codeListCache = new HashMap<>();
+  public static List<Code> selectedCodes = new ArrayList<>();
 
-  /**
-   * Initialize the RandomCodeGenerator with the supplied TerminologyClient.
-   *
-   * @param terminologyClient the TerminologyClient to be used
-   */
-  public static void initialize(TerminologyClient terminologyClient) {
-    RandomCodeGenerator.terminologyClient = terminologyClient;
-    initializeCache();
-  }
-
-  private static void initializeCache() {
-    responseCache = CacheBuilder.newBuilder()
-        .maximumSize(RESPONSE_CACHE_SIZE)
-        .build(
-            new CacheLoader<ExpandInput, ValueSet>() {
-              @Override
-              public ValueSet load(@Nonnull ExpandInput key) {
-                return terminologyClient
-                    .expand(new UriType(key.getValueSetUri()), new IntegerType(EXPAND_PAGE_SIZE),
-                        new IntegerType(key.getOffset()));
-              }
-            }
-        );
-  }
-
-  /**
-   * Clear the terminology service configuration.
-   */
-  public static void reset() {
-    RandomCodeGenerator.terminologyClient = null;
-  }
+  public static RestTemplate restTemplate = new RestTemplate();
 
   /**
    * Gets a random code from the expansion of a ValueSet.
    *
-   * @param valueSetUri the URI of the ValueSet
-   * @param seed a random seed to ensure reproducibility of this result
+   * @param valueSetUri
+   *          the URI of the ValueSet
+   * @param seed
+   *          a random seed to ensure reproducibility of this result
    * @return the randomly selected Code
    */
-  public static Code getCode(String valueSetUri, long seed) {
-    if (terminologyClient == null) {
-      throw new RuntimeException(
-          "Unable to generate code from ValueSet URI: terminology service not configured");
+  @SuppressWarnings("unchecked")
+  public static Code getCode(String valueSetUri, long seed, Code code) {
+    if (urlValidator(valueSetUri)) {
```

Comment:
@hankwallace, now we are validating uri prior to generating Code, if uri is not valid one we are returning default code.(last parameter for the function.)